lib/psm.c

```diff
@@ -511,13 +511,14 @@ void rpmpsmNotify(rpmpsm psm, int what, rpm_loff_t amount)
  */
 static void markReplacedInstance(rpmts ts, rpmte te)
 {
+    /* this must match rpmNameVersionCompare in depends.c */
```

Comment:
I think you meant checkProblems() in transaction.c

Reply:
> We do have rpmNameVersionCompare in depends.c. See the nameversioncompare.diff patch...

(encountered when looking at something else)

The name comparison in that patch has been redundant for a long long time BTW. It used to be necessary back in the days of implicit obsoletes of rpm < 4.4.2.1 when the iterator walked provides, but unnecessary ever since commit 036e9e6dcb8165bb5862b55c6e8825161d78775c. The arch part is more complicated of course.